pkg/ebpf/tcp_dns_layer.go

```diff
@@ -0,0 +1,52 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"encoding/binary"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+// source: https://github.com/weaveworks/scope/blob/master/probe/endpoint/dns_snooper.go
+// Gopacket doesn't provide direct support for DNS over TCP, see https://github.com/google/gopacket/issues/236
+
+type tcpLayer = layers.TCP
```

Comment:
Is there a reason for this type alias? I think you can simply do

```
type tcpWithDNSSupport struct {
	layers.TCP
}
```

Reply:
When I did that, I was getting compile error when doing `m.layers.TCP.LayerPayload()` (line 25). That is why I opted for alias. Please suggest if there is an easier workaround.

Reply:
oh that's because you should call `m.TCP.LayerPayload()` instead

Reply:
Thanks. Will update the PR.